heuristics.py

```diff
@@ -0,0 +1,117 @@
+import random
+
+from kubernetes import client, config, watch
+
+from aws_metrics import get_volatility_metrics
+
+SPOT_LABEL = 'spot-instance'
+VOLATILITY_SPAN = 168
+
+
+config.load_kube_config('./config-spark-on-cat')
+v1 = client.CoreV1Api()
+
+def nodes_available():
+    ready_nodes = []
+    for node in v1.list_node().items:
+        for status in node.status.conditions:
+            if status.status == "True" and status.type == "Ready":
+                ready_nodes.append(node)
+    return ready_nodes
+
+
+def get_available_spot_nodes():
+    available_nodes = nodes_available()
+    spot_nodes = []
+    for node in available_nodes:
+        if SPOT_LABEL in node.metadata.labels and node.metadata.labels[SPOT_LABEL] == 'true':
+            spot_nodes.append(node)
+    return spot_nodes
+
+
+def get_available_persistent_nodes():
+    available_nodes = nodes_available()
+    persistent_nodes = []
+    for node in available_nodes:
+        if SPOT_LABEL not in node.metadata.labels or node.metadata.labels[SPOT_LABEL] == 'false':
+            persistent_nodes.append(node)
+    return persistent_nodes
+
+
+def choose_random_node(nodes):
```

Comment:
Some of your heuristics might want to override this method, I guess - maybe if you were packing the driver nodes tightly onto one node at a time (for some reason?) - do you have any heuristics that would need to do that or will they all choose randomly from the filtered set?